tests/python_package_test/test_engine.py

```diff
@@ -48,2591 +49,2655 @@ def categorize(continuous_x):
-        params = {'objective': 'regression_l2', 'metric': 'rmse'}
-        evals_result = {}
-        gbm = lgb.train(params, tmp_dat_train, num_boost_round=20,
-                        valid_sets=[tmp_dat_train, tmp_dat_val],
-                        verbose_eval=False, evals_result=evals_result)
-        self.assertEqual(len(evals_result['training']['rmse']), 20)
-        self.assertEqual(len(evals_result['valid_1']['rmse']), 20)
-
-    def test_contribs(self):
-        X, y = load_breast_cancer(return_X_y=True)
-        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
-        params = {
-            'objective': 'binary',
-            'metric': 'binary_logloss',
-            'verbose': -1,
-        }
         lgb_train = lgb.Dataset(X_train, y_train)
-        gbm = lgb.train(params, lgb_train, num_boost_round=20)
-
-        self.assertLess(np.linalg.norm(gbm.predict(X_test, raw_score=True)
-                                       - np.sum(gbm.predict(X_test, pred_contrib=True), axis=1)), 1e-4)
-
-    def test_contribs_sparse(self):
-        n_features = 20
-        n_samples = 100
-        # generate CSR sparse dataset
-        X, y = make_multilabel_classification(n_samples=n_samples,
-                                              sparse=True,
-                                              n_features=n_features,
-                                              n_classes=1,
-                                              n_labels=2)
-        y = y.flatten()
-        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
-        params = {
-            'objective': 'binary',
+        gbm_template = lgb.train(params, lgb_train, num_boost_round=10, init_model=init_model)
+        return gbm_template if return_model else mean_squared_error(y_test, gbm_template.predict(X_test))
+
+    gbm = train_and_predict(return_model=True)
+    ret_origin = train_and_predict(init_model=gbm)
+    other_ret = []
+    gbm.save_model('lgb.model')
+    with open('lgb.model') as f:  # check all params are logged into model file correctly
+        assert f.read().find("[num_iterations: 10]") != -1
+    other_ret.append(train_and_predict(init_model='lgb.model'))
+    gbm_load = lgb.Booster(model_file='lgb.model')
+    other_ret.append(train_and_predict(init_model=gbm_load))
+    other_ret.append(train_and_predict(init_model=copy.copy(gbm)))
+    other_ret.append(train_and_predict(init_model=copy.deepcopy(gbm)))
+    with open('lgb.pkl', 'wb') as f:
+        pickle.dump(gbm, f)
+    with open('lgb.pkl', 'rb') as f:
+        gbm_pickle = pickle.load(f)
+    other_ret.append(train_and_predict(init_model=gbm_pickle))
+    gbm_pickles = pickle.loads(pickle.dumps(gbm))
+    other_ret.append(train_and_predict(init_model=gbm_pickles))
+    for ret in other_ret:
+        assert ret_origin == pytest.approx(ret)
+
+
+@unittest.skipIf(not lgb.compat.PANDAS_INSTALLED, 'pandas is not installed')
```

Comment:
could this and other `unittest.skipIf`  in this PR use `pytest.importorskip` like you did in #3798 ? So that we don't need to `import unittest`